src/router/index.js

```diff
@@ -0,0 +1,128 @@
+import {getRoutableSignals, flattenConfig} from './utils'
+import urlMapper from 'url-mapper'
+
+let addressbar
+try {
+  addressbar = require('addressbar')
+} catch (e) {
+  addressbar = {
+    pathname: '/',
+    value: '',
+    origin: '',
+    on() {},
+    removeListener() {}
+  }
+}
+
+export default function Router(options = {}) {
```

Comment:
I see your point and even see a benefits router got as internal module. But I still want to decouple url-mapper for a few reasons. URLs is very important part of web application and must be a stable. It means even "bug fix" or little cosmetic change in mapper could lead to major problems for cerebral users. And it would be a very strange to bump cerebrals major version because of changes in optional module. And of course there is a great chance to miss this bump as well.
It's not a "might be" use case. I have postponed plans on url-mapper which also needs new major on URLON.

> If we want to add a different router that should be the whole router, not just an option. 

Disagreed. Changing mapper is most likely the only change needed. I know several users using router with custom mapper already (including @garth). `OtherRouter` would have copy pasted blocks from original one. It also means that special router module interface must be treated as public.